### PR TITLE
Draft: moved nightscout code into separated folder

### PR DESCRIPTION
**Pull request by 3timeslazy**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -274,53 +253,39 @@ export async function getLibreLinkUpConnection(): Promise<string | null>
...
-    if (!response.data || response.data.length === 0)
-    {
-        return null;
-    }
-    return new Date(response.data.pop().dateString);
+let nightscoutClient: NightscoutAPI;
+if (process.env.NIGHTSCOUT_API_V3 === "true") {
+    nightscoutClient = new ClientV3();
+} else {
+    nightscoutClient = new ClientV1(
```
Comment (ziflex) [inline src/index.ts]: What if the token is not provided?

Comment (ziflex) [reply src/index.ts]: I suggest to add parameters sanitation and validation here.

File: src/nightscout/apiv1.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { Entry, NightscoutAPI } from './interface';
+import { OutgoingHttpHeaders } from 'http';
+import axios from 'axios';
+
+interface nightscoutHttpHeaders extends OutgoingHttpHeaders {
```
Comment (ziflex) [inline src/nightscout/apiv1.ts]: Types must start with capital letters.

File: src/index.ts (hunk 1 above)
Comment (3timeslazy) [reply src/index.ts]: Done

File: src/nightscout/apiv1.ts (hunk 2 above)
Comment (3timeslazy) [reply src/nightscout/apiv1.ts]: Fixed. Thanks!

**Pull request merged**